streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1067,11 +1067,10 @@ boolean maybeCommit() {
             lastCommitMs = now;
             processStandbyRecords = true;
         } else {
-            final int commitPerRequested = taskManager.maybeCommitActiveTasksPerUserRequested();
-            if (commitPerRequested > 0) {
+            committed = taskManager.maybeCommitActiveTasksPerUserRequested();
+            if (committed > 0) {
                 final long requestCommitLatency = advanceNowAndComputeLatency();
                 commitSensor.record(requestCommitLatency / (double) committed, now);
```

Comment:
The problem was here: `committed` could be zero without this fix.

Reply:
Seems like this should have noticeably screwed up these metrics, do we think this branch is just rarely seen or no one has noticed somehow?

Reply:
I think it's rarely executed. Only if a PAPI user request committing via `context.commit()`, what is expected to be rarely used feature.